src/scipp/core/binning.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+from typing import Dict, List, Optional, Union
+
+from .._scipp import core as _cpp
+from .variable import array, Variable, linspace, arange
+from .bins import bin, histogram
+
+
+def _get_coord(x, name):
+    event_coord = x.bins.meta.get(name) if x.bins is not None else None
+    return x.meta.get(name, event_coord)
+
+
+def _upper_bound(x):
+    import numpy as np
+    bound = x.max()
+    bound.value = np.nextafter(bound.value, np.inf)
+    return bound
+
+
+def _parse_coords_arg(x, name, arg):
+    coord = _get_coord(x, name)
+    if isinstance(arg, int):
+        coord = linspace(name, coord.min(), _upper_bound(coord), num=arg + 1)
+    elif isinstance(arg, Variable):
+        if arg.ndim == 0:
+            start = coord.min()
+            step = arg.to(dtype=start.dtype, unit=start.unit)
+            stop = _upper_bound(coord) + step
+            coord = arange(name, start, stop, step=step)
+        else:
+            # TODO check name in arg.dims
+            # TODO handle 2D coords, name must be used, but current implementation
+            # has no support for this
+            coord = arg
+    return coord
```

Comment:
If `arg` is neither `int` not `Variable`, this returns the existing coord and ignores the value of `arg`. I think it should raise a `TypeError` instead.